### PR TITLE
Add pasteDocument action and test

### LINKED ISSUE #48: Paste should respect decoration
For now, paste is adding the content and using the decoration of the piece where it is pasted to.

We'll need some API for this, in order to allow different paste options: paste everything, paste content (with target format), paste content (with source format), ...

### LINKED ISSUE #98: Add support for clipboard copy/paste of faceModel
Clipboard content can be set with part of the underlying FaceModel under a given selection that is copied, and text and decorations can be serialized to a string that can be later on deserialized when clipboard is pasted.

Follow up of #91

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #98 
Fixes #48

### CODE REVIEW

File: rta/src/main/java/com/gluonhq/richtextarea/viewmodel/RichTextAreaViewModel.java [hunk 1]
```diff
@@ -735,10 +740,12 @@ private void updateProperties() {
...
     }
 
-    private Document getCurrentDocument() {
+    private Document getCurrentDocument(Selection selection) {
         // text and indices should be based on the exportable text
         int caret = getTextBuffer().getText(0, getCaretPosition()).length();
-        return new Document(getTextBuffer().getText(), getTextBuffer().getDecorationModelList(), caret);
+        int start = selection.isDefined() ? selection.getStart() : 0;
+        int end = selection.isDefined() ? selection.getEnd() : getTextLength();
+        return new Document(getTextBuffer().getText(start, end), getTextBuffer().getDecorationModelList(start, end), caret);
```
Comment (abhinayagarwal) [inline rta/src/main/java/com/gluonhq/richtextarea/viewmodel/RichTextAreaViewModel.java:748]: There are 3 calls to `getTextBuffer()` in this method. Wondering if we should instead create a variable.

**Review by abhinayagarwal — COMMENTED** [review]
There is another strange behaviour, where additional lines are created while pasting:

https://github.com/gluonhq/rich-text-area/assets/3197675/8b2a0e83-2764-4114-b112-ae8f1d23ccad

**Comment by jperedadnr** [comment]
I've seen that too, before this PR (as there is no decoration involved in your test). When pasting several lines somehow the width of the cell in-between two lines is changed, and defaults to 13 points or so. 
In any case, that should be part of a different issue, I guess.

File: rta/src/main/java/com/gluonhq/richtextarea/viewmodel/RichTextAreaViewModel.java (hunk 1 above)
Comment (jperedadnr) [reply rta/src/main/java/com/gluonhq/richtextarea/viewmodel/RichTextAreaViewModel.java:761]: Not needed. In this class `getTextBuffer()` it is used 30+ times, without extracting variables. After all, it is just a getter from a property, with a null check.

**Review by abhinayagarwal — APPROVED** [review]

**Pull request merged**